plugins/pulp_rpm/yum_plugin/util.py

```diff
@@ -173,3 +175,40 @@ def generate_listing_files(root_publish_dir, repo_publish_dir):
 
         # work up the directory structure
         working_dir = os.path.dirname(working_dir)
+
+
+def errata_format_to_datetime(datetime_str, msg):
+    """
+    Convert known errata date-time formats to datetime object.
+
+    Expected formats are:
+     - '%Y-%m-%d %H:%M:%S'
+     - '%Y-%m-%d %H:%M:%S UTC'
+
+    :param datetime_str: date and time in errata specific format
+    :type  datetime_str: str
+
+    :param msg: additional error message in case of exception
+    :type  msg: str
+
+    :return: parsed date and time
+    :rtype: datetime.datetime
+    :raises ValueError: if the date and time are in unknown format
+    """
+    strptime_pattern = '%Y-%m-%d %H:%M:%S'
+    match_and_drop_utc_regexp = r'^\s*(?P<datetime>.*?)( UTC)?\s*$'
```

Comment:
I was going to write a more specific regex to match the `%Y-%m-%d %H:%M:%S` pattern, but then I decided to suggest that it would be easier to handle without regex. Stripping whitespace can be done with the `datetime_str.strip()' command and removing the optional ` UTC` can be done with:

```
datetime_str = datetime_str.strip()
if datetime_str.endswith(' UTC'):
    datetime_str = datetime_str[:-4]
```